core/src/main/java/com/oath/oak/EntrySet.java

```diff
@@ -169,87 +182,66 @@
+         * being deleted, marked just off-heap).
          */
-        Slice valueSlice;
-        /**
-         * Set to true when entry is found deleted, but not yet suitable to be reused.
+        DELETED,
+
+        /*
+         * When entry is marked deleted, but not yet suitable to be reused.
          * Deletion consists of 3 steps: (1) mark off-heap deleted (LP),
          * (2) CAS value reference to invalid, (3) CAS value version to negative.
          * If not all three steps are done entry can not be reused for new insertion.
          */
-        boolean isFinalizeDeletionNeeded;
-        /**
-         * valueReference is composed of 3 numbers: block ID, value position and value length. All these numbers are
-         * squashed together into a long using the VALUE masks, shifts and indices.
-         * When it equals to {@code INVALID_VALUE_REFERENCE} is means that there is no value referenced from entryIndex.
-         * This field is usually used for CAS purposes since it sits in each entry.
+        DELETED_NOT_FINALIZED,
+
+        /*
+         * There is any entry with the given key and its value is valid.
+         * valueSlice is pointing to the location that is referenced by valueReference.
          */
-        final long valueReference;
-        int entryIndex;
-        /**
-         * This is the version of the value referenced by {@code valueReference}.
-         * If {@code valueReference == INVALID_VALUE_REFERENCE}, then:
-         * {@code version <= INVALID_VERSION} if the removal was completed.
-         * {@code version > INVALID_VERSION} if the removal was not completed.
-         * else
-         * {@code version <= INVALID_VERSION} if the insertion was not completed.
-         * {@code version > INVALID_VERSION} if the insertion was completed.
+        VALID,
+
+        /*
+         * When value is connected to entry, first the value reference is CASed to the new one and after
+         * the value version is set to the new one (written off-heap). Inside entry, when value reference
+         * is invalid its version can only be invalid (0) or negative. When value reference is valid and
+         * its version is either invalid (0) or negative, the insertion or deletion of the entry wasn't
+         * accomplished, and needs to be accomplished.
          */
-        int version;
-        /* key reference, build differently than value reference */
-        final long keyReference;
-
-        LookUp(Slice valueSlice, long valueReference, int entryIndex, int version,
-            long keyReference, boolean isFinalizeDeleteNeeded) {
-            this.valueSlice = valueSlice;
-            this.valueReference = valueReference;
-            this.entryIndex = entryIndex;
-            this.version = version;
-            this.keyReference = keyReference;
-            this.isFinalizeDeletionNeeded = isFinalizeDeleteNeeded;
+        VALID_INSERT_NOT_FINALIZED;
```

Comment:
I agree that it SHOULD be updated.
Currently, I did not change the internal logic, i.e., if the original code updated the `LookUp`/`OpData`, then `ThreadContext` is updated. Otherwise, it is not updated.

Reply:
I will update it in `deleteValueFinish()` and `writeValueFinish()`.